mesonbuild/modules/pkgconfig.py

```diff
@@ -587,32 +639,30 @@ def parse_variable_list(vardict):
                 variables.append((name, value))
             return variables
 
-        variables = self.interpreter.extract_variables(kwargs, dict_new=True)
-        variables = parse_variable_list(variables)
-        unescaped_variables = self.interpreter.extract_variables(kwargs, argname='unescaped_variables')
-        unescaped_variables = parse_variable_list(unescaped_variables)
+        if isinstance(kwargs['variables'], dict):
+            FeatureNew.single_use('variables as dictionary', '0.56.0', state.subproject, location=state.current_node)
```

Comment:
For declare_dependency I handled this via 
```python
VARIABLES_KW.evolve(since='0.54.0', since_values={list: '0.56.0'}),
```
after making that work in c1aeb2f7e4ef81a046ca28d5dad89bb8fdeefafd.

(It will also need to override the `default={}` if we are making it default to list and use dict as since? Maybe?)

Reply:
This is actually super annoying to deal with, for lots of reasons. Blarg.

Reply:
Actually, it turns out not to be. using since_values actually does exatly what we want, because it doesn't check in the case that the value is the default, so we can set the default value to `{}` and it will not emit a warning at all, unlike the current code